### PR TITLE
Fix(client): For loops not iterating correctly

### LINKED ISSUE #7: [BUG]SCRIPT ERROR: @qb-weed/server/main.lua:56
**Describe the bug**
A clear and concise description of what the bug is. A stranger to qbcore should be able to read your bug report and understand how to reproduce it themselves and understand how the feature should work normally.

**To Reproduce**
1. Go to 'start the server'
2. Click on 'nothing'
4. See the error

**Expected behavior**
Not to see the error

**Screenshots**
image

**Questions (please complete the following information):**
 - When you last updated: [just now]
 - Are you using custom resource? which ones? [none]
 - Have you renamed `qb-` to something custom? [no]

**Additional context**
No

### PR DESCRIPTION
**Pull request by YvanCywan**
**Describe Pull request**
Reverted to the old style of for loop as it wasn't working
If your PR is to fix an issue mention that issue here

**Questions (please complete the following information):**
- Have you personally loaded this code into an updated qbcore project and checked all it's functionality? [no] (Be honest)
- Does your code fit the style guidelines? [yes]
- Does your PR fit the contribution guidelines? [yes]

### CODE REVIEW

File: server/main.lua [hunk 1]
```diff
@@ -53,7 +53,7 @@ end
 local function manageHousePlants()
     while true do
         local housePlants = MySQL.query.await('SELECT * FROM house_plants', {})
-        for i in #housePlants do
+        for i, _ in pairs(housePlants) do
```
Comment (BerkieBb) [inline server/main.lua]: This underscore is useless, same for the other one that was changed

**Review by BerkieBb — CHANGES_REQUESTED** [review]

**Comment by BerkieBb** [comment]
Did you test this?

File: server/main.lua (hunk 1 above)
Comment (D4isDAVID) [inline server/main.lua]:
```suggestion
        for i = 1, #housePlants do
```

File: server/main.lua [hunk 2]
```diff
@@ -86,8 +86,8 @@ end
 local function updatePlantGrowth()
     while true do
         local housePlants = MySQL.query.await('SELECT * FROM house_plants', {})
-        for k in #housePlants do
-            growPlant(housePlants[k])
+        for i, _ in pairs(housePlants) do
```
Comment (D4isDAVID) [inline server/main.lua]:
```suggestion
        for i = 1, #housePlants do
```

**Review by D4isDAVID — CHANGES_REQUESTED** [review]

File: server/main.lua (hunk 1 above)
Comment (YvanCywan) [reply server/main.lua]: Still won't work that way

**Comment by YvanCywan** [comment]
> Did you test this?

Tested for the error today, nothing showing.

**Comment by BerkieBb** [comment]
> > Did you test this?
> 
> Tested for the error today, nothing showing.

Well, you're using a pairs iterator, but not using the pairs

**Comment by YvanCywan** [comment]
> > > Did you test this?
> > 
> > 
> > Tested for the error today, nothing showing.
> 
> Well, you're using a pairs iterator, but not using the pairs

I meant nothing showing as it is fixed, whether or not pairs is used. The pairs is however useless, so have removed them in this case.

**Comment by BerkieBb** [comment]
> > > > Did you test this?
> > > 
> > > 
> > > Tested for the error today, nothing showing.
> > 
> > 
> > Well, you're using a pairs iterator, but not using the pairs
> 
> I meant nothing showing as it is fixed, whether or not pairs is used. The pairs is however useless, so have removed them in this case.

I think if you test it now you'll encounter an error

**Comment by YvanCywan** [comment]
> > > > > Did you test this?
> > > > 
> > > > 
> > > > Tested for the error today, nothing showing.
> > > 
> > > 
> > > Well, you're using a pairs iterator, but not using the pairs
> > 
> > 
> > I meant nothing showing as it is fixed, whether or not pairs is used. The pairs is however useless, so have removed them in this case.
> 
> I think if you test it now you'll encounter an error

Nope, starts without issue

**Comment by YvanCywan** [comment]
> > > > > Did you test this?
> > > > 
> > > > 
> > > > Tested for the error today, nothing showing.
> > > 
> > > 
> > > Well, you're using a pairs iterator, but not using the pairs
> > 
> > 
> > I meant nothing showing as it is fixed, whether or not pairs is used. The pairs is however useless, so have removed them in this case.
> 
> I think if you test it now you'll encounter an error

Urite, the commit has been reverted

File: server/main.lua (hunk 1 above)
Comment (D4isDAVID) [reply server/main.lua]: Why not?

Comment (D4isDAVID) [reply server/main.lua]: Your previous commits are different from my suggestion

**Comment by BerkieBb** [comment]
I will test it in a bit and make a suggestion on what it should be

File: server/main.lua (hunk 1 above)
Comment (YvanCywan) [reply server/main.lua]: It now works, committing it now, unsure to why when I tested it with the initial value before (not in my commits). Probably an error with how I tested it to begin with.

I'll go back to feeling like a moron now.

**Comment by BerkieBb** [comment]
I really hadn't picked up on that lol, but something optional would be adding spacing in between i=1

**Comment by YvanCywan** [comment]
> I will test it in a bit and make a suggestion on what it should be

@D4isDAVID solution works, not sure why it threw an error earlier. But it has no issues now.

**Comment by BerkieBb** [comment]
> > I will test it in a bit and make a suggestion on what it should be
> 
> @D4isDAVID solution works, not sure why it threw an error earlier. But it has no issues now.

The `in` keyword only works for functions like each or pairs, not with a numeric iterator

**Comment by YvanCywan** [comment]
> The `in` keyword only works for functions like each or pairs, not with a numeric iterator

It's a good thing I did not test it with ```in``` earlier.

File: server/main.lua [hunk 3]
```diff
@@ -86,8 +86,9 @@ end
 local function updatePlantGrowth()
     while true do
         local housePlants = MySQL.query.await('SELECT * FROM house_plants', {})
-        for k in #housePlants do
-            growPlant(housePlants[k])
+        for i=1, #housePlants do
+            print(i)
```
Comment (D4isDAVID) [inline server/main.lua]:
```suggestion
```
Print statement :)

**Review by D4isDAVID — CHANGES_REQUESTED** [review]

**Review by D4isDAVID — APPROVED** [review]
LGTM

**Review by BerkieBb — APPROVED** [review]

**Pull request merged**